core/src/test/java/org/elasticsearch/index/get/GetFieldTests.java

```diff
@@ -0,0 +1,192 @@
+
+    public void testToAndFromXContent() throws Exception {
+        XContentType xContentType = randomFrom(XContentType.values());
+        Tuple<GetField, GetField> tuple = randomGetField(xContentType);
+        GetField getField = tuple.v1();
+        GetField expectedGetField = tuple.v2();
+        BytesReference originalBytes = toXContent(getField, xContentType, true);
+        //test that we can parse what we print out
+        GetField parsedGetField;
+        try (XContentParser parser = createParser(xContentType.xContent(), originalBytes)) {
+            //we need to move to the next token, the start object one that we manually added is not expected
+            assertEquals(XContentParser.Token.START_OBJECT, parser.nextToken());
+            assertEquals(XContentParser.Token.FIELD_NAME, parser.nextToken());
+            parsedGetField = GetField.fromXContent(parser);
+            assertEquals(XContentParser.Token.END_ARRAY, parser.currentToken());
+            assertEquals(XContentParser.Token.END_OBJECT, parser.nextToken());
+            assertNull(parser.nextToken());
+        }
+        assertEquals(expectedGetField, parsedGetField);
+        BytesReference finalBytes = toXContent(parsedGetField, xContentType, true);
+        assertSameOutput(originalBytes, finalBytes, xContentType);
+    }
+
+    private static GetField copyGetField(GetField getField) {
+        return new GetField(getField.getName(), getField.getValues());
+    }
+
+    private static GetField mutateGetField(GetField getField) {
+        List<Supplier<GetField>> mutations = new ArrayList<>();
+        mutations.add(() -> new GetField(randomUnicodeOfCodepointLength(15), getField.getValues()));
+        mutations.add(() -> new GetField(getField.getName(), randomGetField(XContentType.JSON).v1().getValues()));
+        return randomFrom(mutations).get();
+    }
+
+    public static Tuple<GetField, GetField> randomGetField(XContentType xContentType) {
+        if (randomBoolean()) {
+            String fieldName = randomFrom(ParentFieldMapper.NAME, RoutingFieldMapper.NAME, UidFieldMapper.NAME);
+            GetField getField = new GetField(fieldName, Collections.singletonList(randomAsciiOfLengthBetween(3, 10)));
+            return Tuple.tuple(getField, getField);
+        }
+        String fieldName = randomAsciiOfLengthBetween(3, 10);
+        Tuple<List<Object>, List<Object>> tuple = randomStoredFieldValues(xContentType);
+        GetField input = new GetField(fieldName, tuple.v1());
+        GetField expected = new GetField(fieldName, tuple.v2());
+        return Tuple.tuple(input, expected);
+    }
+
+    //TODO move this to some utility class
+    /**
+     * Asserts that the provided {@link BytesReference}s hold the same content. The comparison is done between the map
+     * representation of the provided objects.
+     */
+    public static void assertSameOutput(BytesReference expected, BytesReference actual, XContentType xContentType) throws IOException {
```

Comment:
I think we can remove the TODO; rename the method to something like `assertEqualsAsMap`, and change it to

```
        try (
                XContentParser expectedParser = xContentType.xContent().createParser(expected);
                XContentParser actualParser = xContentType.xContent().createParser(actual);
        ) {
            assertEquals(expectedParser.map(), actualParser.map());
        }
```